shortcut.js

```diff
@@ -44,10 +102,16 @@ const setupPrPage = async () => {
     btn.innerText = "Run nixpkgs-review";
     actions.prepend(btn);
     btn.onclick = () => {
-      window.open(`https://github.com/${repo}/actions/workflows/review.yml#${pr}`);
+      const params = new URLSearchParams({ ...reviewDefaults(getPrDetails(pr)), pr });
+      window.open(`https://github.com/${repo}/actions/workflows/review.yml#dispatch:${params}`);
     };
   }
 
+  const { hasLinuxRebuilds, hasDarwinRebuilds } = getPrDetails(pr);
```

Comment:
If I understand this correctly, `getPrDetails` is run once on page load.

That could be a problem if the labels change while the page is open, e.g. if the page was open for a while without a refresh. This happens more often that you might think.

Would it be better to get the PR details when the button is clicked, instead of when adding the button to the page?

Reply:
> If I understand this correctly, `getPrDetails` is run once on page load.
> 
> That could be a problem if the labels change while the page is open, e.g. if the page was open for a while without a refresh. This happens more often that you might think.

Yes, it is run once on page load, but also whenever anything on the page changes using the MutationObserver below.

> Would it be better to get the PR details when the button is clicked, instead of when adding the button to the page?

This is already the case, see L105. However at least getting the number of rebuilds before clicking the button is still necessary to set the disabled attribute on the button if there are no rebuilds.

Reply:
Ah ok. That makes sense; I forgot about the observer.